test/test_pebble.py

```diff
@@ -1076,6 +1161,423 @@ def test_get_services_names(self):
+        ])
+
+    def test_make_dir_basic(self):
+        self.client.responses.append({
+            "result": [{'path': '/foo/bar'}],
+            'status': 'OK',
+            'status-code': 200,
+            'type': 'sync',
+        })
+        self.client.make_dir('/foo/bar')
+        dir_item = {
+            'path': '/foo/bar',
+        }
+        self.assertEqual(self.client.requests, [
+            ('POST', '/v1/files', None, {'action': 'make-dirs', 'dirs': [dir_item]}),
+        ])
+
+    def test_make_dir_full(self):
+        self.client.responses.append({
+            "result": [{'path': '/foo/bar'}],
+            'status': 'OK',
+            'status-code': 200,
+            'type': 'sync',
+        })
+        self.client.make_dir('/foo/bar', make_parents=True, permissions=0o600,
+                             user=12, group=34)
+
+        dir_item = {
+            'path': '/foo/bar',
+            'make-parents': True,
+            'permissions': '600',
+            'user-id': 12,
+            'group-id': 34,
+        }
+        self.assertEqual(self.client.requests, [
+            ('POST', '/v1/files', None, {'action': 'make-dirs', 'dirs': [dir_item]}),
+        ])
+
+    def test_make_dir_error(self):
+        self.client.responses.append({
+            "result": [{
+                'path': '/foo/bar',
+                'error': {
+                    'kind': 'permission-denied',
+                    'message': 'permission denied',
+                },
+            }],
+            'status': 'OK',
+            'status-code': 200,
+            'type': 'sync',
+        })
+        with self.assertRaises(pebble.PathError) as cm:
+            self.client.make_dir('/foo/bar')
+        self.assertIsInstance(cm.exception, pebble.Error)
+        self.assertEqual(cm.exception.kind, 'permission-denied')
+        self.assertEqual(cm.exception.message, 'permission denied')
+
+    def test_remove_path_basic(self):
+        self.client.responses.append({
+            "result": [{'path': '/boo/far'}],
+            'status': 'OK',
+            'status-code': 200,
+            'type': 'sync',
+        })
+        self.client.remove_path('/boo/far')
+        dir_item = {
+            'path': '/boo/far',
+        }
+        self.assertEqual(self.client.requests, [
+            ('POST', '/v1/files', None, {'action': 'remove', 'paths': [dir_item]}),
+        ])
+
+    def test_remove_path_recursive(self):
+        self.client.responses.append({
+            "result": [{'path': '/boo/far'}],
+            'status': 'OK',
+            'status-code': 200,
+            'type': 'sync',
+        })
+        self.client.remove_path('/boo/far', recursive=True)
+
+        dir_item = {
+            'path': '/boo/far',
+            'recursive': True,
+        }
+        self.assertEqual(self.client.requests, [
+            ('POST', '/v1/files', None, {'action': 'remove', 'paths': [dir_item]}),
```

Comment:
I'm curious why you pulled 'dir_item' out and indirect it, vs including it in the content so that we can see the full request structure.

Reply:
Rejiggered this a little.